src/game/default/bg_pmove.c

```diff
@@ -637,43 +640,44 @@ static void Pm_CheckDuck(void) {
 	const vec_t height = pm->maxs[2] - pm->mins[2];
 
 	if (pm->s.type == PM_DEAD) {
-		pm->s.flags |= PMF_DUCKED;
+
+		if (pm->s.flags & PMF_GIBLET) {
+			VectorClear(pml.view_offset);
+		} else {
+			pm->maxs[2] = pm->mins[2] + height * 0.25;
+			pml.view_offset[2] = pm->mins[2] + height * 0.15;
+		}
+
 	} else {
-		// if on the ground and requesting to crouch, duck
 		if ((pm->s.flags & PMF_ON_GROUND) && pm->cmd.up < 0) {
 			pm->s.flags |= PMF_DUCKED;
-		} else { // stand up if possible
+		} else {
 			cm_trace_t trace = pm->Trace(pm->s.origin, pm->s.origin, pm->mins, pm->maxs);
 			if (trace.all_solid) {
 				pm->s.flags |= PMF_DUCKED;
 			}
 		}
-	}
 
-	if (pm->s.flags & PMF_DUCKED) { // ducked, reduce height
-		vec_t target = pm->mins[2];
+		if (pm->s.flags & PMF_DUCKED) { // ducked, reduce height
+			vec_t target = pm->mins[2] + height * 0.5;
 
-		if (pm->s.type == PM_DEAD)
-			target += height * 0.15;
-		else
-			target += height * 0.5;
+			if (pml.view_offset[2] > target) // go down
+				pml.view_offset[2] -= pml.time * PM_SPEED_DUCK_STAND;
 
-		if (pml.view_offset[2] > target) // go down
-			pml.view_offset[2] -= pml.time * PM_SPEED_DUCK_STAND;
+			if (pml.view_offset[2] < target)
+				pml.view_offset[2] = target;
 
-		if (pml.view_offset[2] < target)
-			pml.view_offset[2] = target;
-
-		// change the bounding box to reflect ducking and jumping
-		pm->maxs[2] = pm->maxs[2] + pm->mins[2] * 0.5;
-	} else {
-		const vec_t target = pm->mins[2] + height * 0.9;
+			// change the bounding box to reflect ducking
+			pm->maxs[2] = pm->maxs[2] + pm->mins[2] * 0.5;
+		} else {
+			const vec_t target = pm->mins[2] + height * 0.9;
 
-		if (pml.view_offset[2] < target) // go up
-			pml.view_offset[2] += pml.time * PM_SPEED_DUCK_STAND;
+			if (pml.view_offset[2] < target) // go up
+				pml.view_offset[2] += pml.time * PM_SPEED_DUCK_STAND;
 
-		if (pml.view_offset[2] > target)
-			pml.view_offset[2] = target;
+			if (pml.view_offset[2] > target)
+				pml.view_offset[2] = target;
+		}
```

Comment:
Simplified the ducking logic such that PM_DEAD is it's own separate case, not mixed in with PM_NORMAL.